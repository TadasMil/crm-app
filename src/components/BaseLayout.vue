<script setup name="BaseLayout" lang="ts">
import { computed } from 'vue'
import { useRoute } from 'vue-router'

const route = useRoute()

const pageName = computed(() => route.meta.pageName)
</script>

<template>
  <div class="bg-secondary">
    <div class="mx-4 space-y-4">
      <h1
        class="border-text-primary flex h-20 w-full items-end border-b border-primary pb-1 text-2xl font-bold uppercase tracking-widest text-primary"
      >
        {{ pageName }}
      </h1>
      <slot />
    </div>
  </div>
</template>
