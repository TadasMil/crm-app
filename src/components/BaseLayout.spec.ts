import { render } from '@testing-library/vue'
import { describe, expect, it, vi } from 'vitest'

import BaseLayout from './BaseLayout.vue'

vi.mock('vue-router', () => ({
  useRoute: vi.fn(() => ({ meta: { pageName: 'Home' } })),
}))

describe('BaseLayout.spec.js', () => {
  it('should render the component with pageName', () => {
    const { html } = render(BaseLayout)

    expect(html()).toMatchSnapshot()
  })
})
