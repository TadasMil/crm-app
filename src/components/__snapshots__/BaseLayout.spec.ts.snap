// Vitest Snapshot v1

exports[`BaseLayout.spec.js > should render the component with pageName 1`] = `
"<div class=\\"bg-secondary\\">
  <div class=\\"mx-4 space-y-4\\">
    <h1 class=\\"border-text-primary flex h-20 w-full items-end border-b border-primary pb-1 text-2xl font-bold uppercase tracking-widest text-primary\\">Home</h1>
  </div>
</div>"
`;
